airflow/contrib/operators/gcp_translate_speech_operator.py

```diff
@@ -0,0 +1,156 @@
+      is required by the API and defaults to
+      the target language of the current instance.
+
+    :type format_: str or None
+    :param format_: (Optional) One of ``text`` or ``html``, to specify
+      if the input text is plain text or HTML.
+
+    :type source_language: str or None
+    :param source_language: (Optional) The language of the text to
+      be translated.
+
+    :type model: str or None
+    :param model: (Optional) The model used to translate the text, such
+      as ``'base'`` or ``'nmt'``.
+
+    :param project_id: Optional, Google Cloud Platform Project ID where the Compute
+        Engine Instance exists.  If set to None or missing, the default project_id from the GCP connection is
+        used.
+    :type project_id: str
+
+    :param gcp_conn_id: Optional, The connection ID used to connect to Google Cloud
+        Platform. Defaults to 'google_cloud_default'.
+    :type gcp_conn_id: str
+    """
+
+    # [START translate_speech_template_fields]
+    template_fields = ('target_language', 'format_', 'source_language', 'model', 'project_id', 'gcp_conn_id')
+    # [END translate_speech_template_fields]
+
+    @apply_defaults
+    def __init__(
+        self,
+        audio,
+        config,
+        target_language,
+        format_,
+        source_language,
+        model,
+        project_id=None,
+        gcp_conn_id='google_cloud_default',
+        *args,
+        **kwargs
+    ):
+        super(GcpTranslateSpeechOperator, self).__init__(*args, **kwargs)
+        self.audio = audio
+        self.config = config
+        self.target_language = target_language
+        self.format_ = format_
+        self.source_language = source_language
+        self.model = model
+        self.project_id = project_id
+        self.gcp_conn_id = gcp_conn_id
+
+    def execute(self, context):
+        _speech_to_text_hook = GCPSpeechToTextHook(gcp_conn_id=self.gcp_conn_id)
+        _translate_hook = CloudTranslateHook(gcp_conn_id=self.gcp_conn_id)
+
+        recognize_result = _speech_to_text_hook.recognize_speech(
+            config=self.config, audio=self.audio
+        )
+        recognize_dict = MessageToDict(recognize_result)
+
+        if len(recognize_dict['results']) == 0:
+            self.log.info("No recognition results")
+            return {}
+
+        try:
+            transcript = recognize_dict['results'][0]['alternatives'][0]['transcript']
+        except KeyError as key:
+            raise AirflowException("Wrong response '{}' returned - it should contain {} field"
+                                   .format(recognize_dict, key))
+
+        try:
+            translation = _translate_hook.translate(
+                values=transcript,
+                target_language=self.target_language,
+                format_=self.format_,
+                source_language=self.source_language,
+                model=self.model
+            )
+            self.log.info('translated output: {}'.format(translation))
```

Comment:
Avoid formatting the string before passing to the logger. In case the message does not need to be logged, the full string still got initialized.
```suggestion
            self.log.info('translated output: %s',translation)
```

Reply:
Changed it to %s as you suggested